tests/func/test_exception.py

```diff
@@ -208,3 +208,73 @@ def code_to_debug():
             session.send_request('continue').wait_for_response(freeze=False)
 
         session.wait_for_exit()
+
+
+@pytest.mark.parametrize('max_frames', ['default', 'all', 10])
+def test_exception_stack(pyfile, run_as, start_method, max_frames):
+    @pyfile
+    def code_to_debug():
+        from dbgimporter import import_and_enable_debugger
+        import_and_enable_debugger()
+        def do_something(n):
+            if n <= 0:
+                raise ArithmeticError('bad code') # @unhandled
+            do_something(n - 1)
+        do_something(100)
+
+    if max_frames == 'all':
+        # trace back compresses repeated text
+        min_expected_lines = 10
+        max_expected_lines = 1000
+        args = {'maxExcpetionStackFrames': 0}
+    elif max_frames == 'default':
+        min_expected_lines = 10
+        max_expected_lines = 1000
+        args = {}
+    else:
+        min_expected_lines = 10
+        max_expected_lines = 21
+        args = {'maxExcpetionStackFrames': 10}
+
+    line_numbers = get_marked_line_numbers(code_to_debug)
+    with DebugSession() as session:
+        session.initialize(
+            target=(run_as, code_to_debug),
+            start_method=start_method,
+            ignore_unobserved=[Event('continued')],
+            expected_returncode=ANY.int,
+            args=args,
+        )
+        session.send_request('setExceptionBreakpoints', {
+            'filters': ['uncaught']
+        }).wait_for_response()
+        session.start_debugging()
+
+        hit = session.wait_for_thread_stopped(reason='exception')
+        frames = hit.stacktrace.body['stackFrames']
+        assert frames[0]['line'] == line_numbers['unhandled']
+
+        resp_exc_info = session.send_request('exceptionInfo', {
+            'threadId': hit.thread_id
+        }).wait_for_response()
+
+        expected = ANY.dict_with({
+            'exceptionId': ANY.such_that(lambda s: s.endswith('ArithmeticError')),
```

Comment:
`pattern.Regex` might come in handy here and below - it prints better in the error output if assert fails.